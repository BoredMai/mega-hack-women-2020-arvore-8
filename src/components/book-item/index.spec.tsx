import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import BookItem from '.';

describe('<BookItem>', () => {
  const book = { icon: 'princesa', label: 'Princesa', url: 'princesa' };
  const props = {
    book,
    isSelected: false,
    onSelectBook: jest.fn().mockName('onSelectBook'),
  };

  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<BookItem {...props} />) as ShallowWrapper;
  });

  it('renders properly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('adds class when book is disabled', () => {
    wrapper.setProps({ book: { ...book, disabled: true } });
    expect(wrapper.find('.bookDisabled')).toHaveLength(1);
  });

  it('adds class when book is selected', () => {
    wrapper.setProps({ isSelected: true });
    expect(wrapper.find('.bookSelected')).toHaveLength(1);
  });

  it('calls onSelectBook on button click', () => {
    wrapper.find('.book').simulate('click');

    expect(props.onSelectBook).toHaveBeenCalledWith(book);
  });
});
