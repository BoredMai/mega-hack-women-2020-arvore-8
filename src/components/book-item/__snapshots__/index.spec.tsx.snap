// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookItem> renders properly 1`] = `
<div
  className="book"
  onClick={[Function]}
>
  <img
    id="book-cover-Princesa"
    src="src/assets/library/princesa.png"
  />
  <label
    htmlFor="book-cover-Princesa"
  >
    Princesa
  </label>
</div>
`;
